app/src/main/java/org/cru/godtools/adapter/ToolsHeaderAdapter.kt

```diff
@@ -0,0 +1,61 @@
+package org.cru.godtools.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import org.ccci.gto.android.common.recyclerview.adapter.DataBindingViewHolder
+import org.cru.godtools.content.ToolHeader
+import org.cru.godtools.databinding.ToolsTutorialBannerBinding
+import org.cru.godtools.fragment.HeaderBannerCallbacks
+
+class ToolsHeaderAdapter internal constructor(builder: Builder) : BaseEmptyListHeaderFooterAdapter(builder) {
+    class Builder : BaseEmptyListHeaderFooterAdapter.Builder<Builder>() {
+        fun build(): ToolsHeaderAdapter {
+            return ToolsHeaderAdapter(this).apply {
+                this.adapterToolHeader = toolHeader
+            }
+        }
+
+        fun toolHeader(header: ToolHeader): Builder {
+            this.toolHeader = header
+            return this
+        }
+
+        lateinit var toolHeader: ToolHeader
+    }
+
+    // region Header
+
+    private lateinit var adapterToolHeader: ToolHeader
+
+    var isHeaderVisible: Boolean = true
+        set(value) {
+            field = value
+            headerAdapter?.notifyDataSetChanged()
+        }
+
+    lateinit var headerBannerCallbacks: HeaderBannerCallbacks
+
+    override fun getHeaderItemCount(): Int {
+        return if (isHeaderVisible) 1 else 0
+    }
+
+    override fun onBindHeaderItemViewHolder(holder: RecyclerView.ViewHolder, localPosition: Int) =
+        (holder as DataBindingViewHolder<ToolsTutorialBannerBinding>).let {
+            it.binding.callback = headerBannerCallbacks
```

Comment:
the callback should be set when the DataBinding is inflated

Reply:
@frett it looks like you completed most of these changes, am I correct?

Reply:
yeah, pending feedback from justin on the banner we are done